### PR TITLE
added steadyNSturb class

### PR DESCRIPTION
**Pull request by saddamhijazi**

### CODE REVIEW

File: src/problems/steadyNSturb/NLsolve.H [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+\*---------------------------------------------------------------------------*/
+
+scalar residual = 1;
+scalar uresidual = 1;
+Vector<double> uresidual_v(0,0,0);
+
+scalar presidual = 1;
+
+// Variable that can be changed
+scalar tolerance = 1e-5;
```
Comment (giovastabile) [inline src/problems/steadyNSturb/NLsolve.H]: change this as in steadyNS case that is read from file

**Comment by ndem0** [comment]
Header guards missing in some header files

**Comment by giovastabile** [comment]
Nicola if you mean NLsolve.H file that is not a real header file. It simply a file a file included from another source file. This is a strange way they have in OpenFOAM to make the code shorter.

**Comment by ndem0** [comment]
@giovastabile I read more carefully the files: so basically you need to call them many times during the program, right? In this case, the missing of header guards is correct, just a bit unconventional, so forget my warning.

**Comment by giovastabile** [comment]
@ndem0 Yes exactly, I know it is unconventional, I took it from the way the implement solvers in OpenFOAM. Sometimes when they have some lines of code doing a particular operation they wrap them in an H file and they include the H file. I guess it's just a workaround they use to make the code shorter and more readable (at least this is their idea).

File: src/problems/steadyNSturb/NLsolve.H [hunk 2]
```diff
@@ -35,51 +35,43 @@ Vector<double> uresidual_v(0,0,0);
 scalar presidual = 1;
 
 // Variable that can be changed
-scalar tolerance = 1e-5;
 turbulence->read();
 std::ofstream res_os;
 res_os.open("./ITHACAoutput/Offline/residuals", std::ios_base::app);
 
-int maxIter = 3000;
```
Comment (giovastabile) [inline src/problems/steadyNSturb/NLsolve.H]: I think we could leave this variable but I would also read it from file and use to exit the loop if the number of iterations is bigger than the prescribed value. Moreover I would add it also in the steadyNS class.

File: src/problems/steadyNSturb/steadyNSturb.C [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+	
+
+	for (label i = 0; i < Csize; i++)
+	{
+
+		Info << "Filling layer number " << i+1 << " in the matrix CT1_matrix" << endl;
+
+		for (label j = 0; j < Nnutmodes; j++)
+		{
+			for (label k = 0;k < Csize; k++)
```
Comment (mtezzele) [inline src/problems/steadyNSturb/steadyNSturb.C:143]: space after ;

File: src/problems/steadyNSturb/steadyNSturb.C [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+
+	
+
+	for (label i = 0; i < Csize; i++)
+	{
+		Info << "Filling layer number " << i+1 << " in the matrix CT2_matrix" << endl;
+
+		for (label j = 0; j < Nnutmodes; j++)
+		{
+			for (label k = 0;k < Csize; k++)
```
Comment (mtezzele) [inline src/problems/steadyNSturb/steadyNSturb.C:210]: space after ;

File: .gitignore [hunk 5]
```diff
@@ -1,4 +1,5 @@
 **/ITHACAoutput
+working_examples
```
Comment (mtezzele) [inline .gitignore:2]: ??

**Review by mtezzele — COMMENTED** [review]
I would suggest to check the indentation of all the files. For example steadyNSturb/NLsolve.H has some indentation issues.

**Pull request merged**